### PR TITLE
*: move leak check from each test to suite

### PR DESCRIPTION
**Pull request by tiancaiamao**
Fix CI in unit test:

```
FAIL: tables_test.go:313: testSuite.TestTableFromMeta

tables_test.go:332:
    ...
/home/jenkins/workspace/tidb_ghpr_unit_test/go/src/github.com/pingcap/tidb/util/testleak/leaktest.go:111:
    c.Errorf(“Test appears to have leaked: %v”, g)
... Error: Test appears to have leaked: github.com/pingcap/tidb/store/tikv/tikvrpc.CheckStreamTimeoutLoop(0xc4201df2c0)
    /home/jenkins/workspace/tidb_ghpr_unit_test/go/src/github.com/pingcap/tidb/store/tikv/tikvrpc/tikvrpc.go:470 +0x1b0
created by github.com/pingcap/tidb/store/mockstore/mocktikv.NewRPCClient
    /home/jenkins/workspace/tidb_ghpr_unit_test/go/src/github.com/pingcap/tidb/store/mockstore/mocktikv/rpc.go:449 +0x5c
```

@coocood

### CODE REVIEW

**Comment by shenli** [comment]
LGTM

**Comment by coocood** [comment]
@tiancaiamao 
Seems there are still many leak check in test function.
Can we finish all the moving work at once?

**Comment by tiancaiamao** [comment]
I've go over the test code again, check all the leak test that maybe false positive, and made some tiny update.

@shenli @coocood

File: util/auth/auth_test.go [hunk 1]
```diff
@@ -24,20 +23,17 @@ type testAuthSuite struct {
 }
 
 func (s *testAuthSuite) TestEncodePassword(c *C) {
-	defer testleak.AfterTest(c)()
```
Comment (shenli) [inline util/auth/auth_test.go]: Add leak check in testAuthSuite.SetUp/TearDown?

**Comment by tiancaiamao** [comment]
PTAL @shenli

**Pull request merged**